src/main/java/com/aws/greengrass/deployment/DeploymentConfigMerger.java

```diff
@@ -190,38 +190,63 @@ public static void waitForServicesToStart(Collection<GreengrassService> services
             throws InterruptedException, ServiceUpdateException {
         // Relying on the fact that all service lifecycle steps should have timeouts,
         // assuming this loop will not get stuck waiting forever
-        while (true) {
-            boolean allServicesRunning = true;
-            for (GreengrassService service : servicesToTrack) {
-                State state = service.getState();
-
-                // If a service is previously BROKEN, its state might have not been updated yet when this check
-                // executes. Therefore we first check the service state has been updated since merge map occurs.
-                if (service.getStateModTime() > mergeTime && State.BROKEN.equals(state)) {
-                    logger.atWarn(MERGE_CONFIG_EVENT_KEY).kv(SERVICE_NAME_LOG_KEY, service.getName())
-                            .log("merge-config-service BROKEN");
-                    throw new ServiceUpdateException(
-                            String.format("Service %s in broken state after deployment", service.getName()),
-                            DeploymentErrorCode.COMPONENT_BROKEN,
-                            DeploymentErrorCodeUtils.classifyComponentError(service, kernel));
-                }
-                if (!service.reachedDesiredState()) {
-                    allServicesRunning = false;
-                    continue;
-                }
-                if (State.RUNNING.equals(state) || State.FINISHED.equals(state) || !service.shouldAutoStart()
-                        && service.reachedDesiredState()) {
-                    continue;
-                }
-                allServicesRunning = false;
-            }
-            if (allServicesRunning) {
+        while (!isAllServiceInDesiredState(servicesToTrack, mergeTime, kernel)) {
+            Thread.sleep(WAIT_SVC_START_POLL_INTERVAL_MILLISEC); // hardcoded
+        }
+    }
+
+    /**
+     * Completes the provided future when all the listed services are running.
+     * Exits early if the future is cancelled
+     *
+     * @param servicesToTrack       services to track
+     * @param mergeTime             time the merge was started, used to check if a service is broken due to the merge
+     * @param kernel                kernel
+     * @param totallyCompleteFuture used to check if a deployment is cancelled
+     * @throws InterruptedException   if the thread is interrupted while waiting here
+     * @throws ServiceUpdateException if a service could not be updated
+     */
+    public static void waitForServicesToStart(Collection<GreengrassService> servicesToTrack, long mergeTime,
```

Comment:
Why are we overloading `waitForServicesToStart`?
We should ensure that a single abstraction for this method exists for all deployment types.

Reply:
It's overloaded for now because Kernel Update Deployment Task calls it too.
I wanted to keep this PR strictly related to Default Deployment.
I will add a TODO comment and fix it in the next PR with Kernel Update Deployment Task changes.